livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/llm.py

```diff
@@ -62,6 +62,45 @@ def __init__(
         )
         self._running_fncs: MutableSet[asyncio.Task[Any]] = set()
 
+    @staticmethod
+    def with_fireworks(
+        *,
+        model: str = "accounts/fireworks/models/llama-v3p1-70b-instruct",
+        api_key: str | None = None,
+        base_url: str | None = "https://api.fireworks.ai/inference/v1",
+        client: openai.AsyncClient | None = None,
+    ) -> LLM:
+        return LLM(model=model, api_key=api_key, base_url=base_url, client=client)
+
+    @staticmethod
+    def with_groq(
+        *,
+        model: str | GroqChatModels = "llama3-8b-8192",
+        api_key: str | None = None,
+        base_url: str | None = "https://api.groq.com/openai/v1",
+        client: openai.AsyncClient | None = None,
+    ) -> LLM:
+        return LLM(model=model, api_key=api_key, base_url=base_url, client=client)
+
+    @staticmethod
+    def with_ollama(
+        *,
+        model: str = "llama3.1",
+        base_url: str | None = "http://localhost:11434/v1",
+        client: openai.AsyncClient | None = None,
+    ) -> LLM:
+        return LLM(model=model, api_key="ollama", base_url=base_url, client=client)
+
+    @staticmethod
+    def with_perplexity(
+        *,
+        model: str | PerplexityChatModels = "llama-3.1-sonar-small-128k-chat",
+        api_key: str | None = None,
+        base_url: str | None = "https://api.perplexity.ai",
+        client: openai.AsyncClient | None = None,
+    ) -> LLM:
+        return LLM(model=model, api_key=api_key, base_url=base_url, client=client)
+
     @staticmethod
     def create_azure_client(
```

Comment:
It may not be the best name we picked for Azure, but should we keep the names consistent? cc @livekit/agent-devs

Reply:
i think an important difference here is that Azure is still an OpenAI usecase, while the rest are other services supporting the same API. so `create_perplexity_client` makes less sense than it does for azure imo. i'm fine with changing it if there's consensus though

Reply:
got it. just another potential issue: do we go with `ai` at the end of the names? or not? or depending on what the provider calls itself? for example `groq`/`fireworks` vs `octo`/`together`

Reply:
I think without is better. AI is kind of redundant. let's also alias the `create_azure_client` to `with_azure`, while marking the former deprecated.